pkg/manifests/nmstateconfig.go

```diff
@@ -60,9 +61,18 @@ func ProcessNMStateConfig(infraEnv aiv1beta1.InfraEnv) ([]*models.HostStaticNetw
 	}
 
 	var staticNetworkConfig []*models.HostStaticNetworkConfig
+	networkYaml, err := json.Marshal(nmStateConfig.Spec.NetConfig.Interfaces)
+	if err != nil {
+		return nil, err
+	}
 	staticNetworkConfig = append(staticNetworkConfig, &models.HostStaticNetworkConfig{
 		MacInterfaceMap: buildMacInterfaceMap(nmStateConfig),
-		NetworkYaml:     string(nmStateConfig.Spec.NetConfig.Raw),
+		NetworkYaml:     string(networkYaml),
 	})
 	return staticNetworkConfig, nil
 }
+
+func GetNodeZeroIP() string {
+	nmStateConfig := getNMStateConfig()
+	return nmStateConfig.Spec.NetConfig.Interfaces[0].IPV4.Address[0].IP
```

Comment:
I think that's it not going to be possible to change the Raw field to specific fields in Netconfig (in https://github.com/openshift/assisted-service/pull/3637)  because of all the additional fields that Interfaces can potentially have - bonds, vlans, bridges e.g. https://nmstate.io/examples.html. Maybe just change this to do a string find for "interfaces", then first "address".

Reply:
The RawNetConfig is a set of bytes, so you could pass those bytes to yaml.Unmarshall() with a data structure that includes only the fields you need to extract. You don't need to change the field in the upstream API and you don't have to fully describe all of the possible fields in the struct you unmarshall into.